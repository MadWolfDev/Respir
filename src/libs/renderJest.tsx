import { ThemeProvider } from '@mui/material/styles';
import { theme } from './theme/Theme';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

export const renderJest = (children: JSX.Element) => {
  return render(
    <ThemeProvider theme={theme}>
      <BrowserRouter>{children}</BrowserRouter>
    </ThemeProvider>
  );
};
