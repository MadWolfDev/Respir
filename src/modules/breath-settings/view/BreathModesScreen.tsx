import { styled } from '@mui/material/styles';
import { useBreathModeStore } from '../domain/breathModeStore';
import { motion } from 'framer-motion';
import { Dispatch, SetStateAction } from 'react';
import { Display } from './WelcomeScreen';
import { BreathModes } from '../infra/defaultBreathConfigs';
import { BreathModesDisplay } from '../domain/BreathModesDisplay.type';
import { CustomButton } from '../../../libs/design-system/CustomButton';
import { createButtonsExitAnim } from './animations/button/createButtonExitAnim';
import { createButtonsVariantAnim } from './animations/button/createButtonsVariantAnim';

export const BreathModesScreen = ({
  setCurrentDisplay,
}: {
  setCurrentDisplay: Dispatch<SetStateAction<Display>>;
}) => {
  const updateBreathModeSelection = useBreathModeStore(
    (state) => state.updateBreathMode
  );
  const changeBreathMode = (newBreathMode: BreathModes) => {
    updateBreathModeSelection(newBreathMode);
    setCurrentDisplay(Display.slidersDisplay);
  };

  const buttonsExitAnims = createButtonsExitAnim(5);
  const buttonsAnimVariants = createButtonsVariantAnim(5);

  return (
    <ButtonContainer>
      <CustomButton
        buttonText={BreathModesDisplay.custom}
        handleClick={() => changeBreathMode('custom')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[4]}
        exitAnim={buttonsExitAnims[4]}
      />

      <CustomButton
        buttonText={BreathModesDisplay.square}
        handleClick={() => changeBreathMode('square')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[3]}
        exitAnim={buttonsExitAnims[3]}
      />

      <CustomButton
        buttonText={BreathModesDisplay.relaxation}
        handleClick={() => changeBreathMode('relaxation')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[2]}
        exitAnim={buttonsExitAnims[2]}
      />

      <CustomButton
        buttonText={BreathModesDisplay.vitality}
        handleClick={() => changeBreathMode('vitality')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[1]}
        exitAnim={buttonsExitAnims[1]}
      />

      <CustomButton
        buttonText={BreathModesDisplay.heartCoherence}
        handleClick={() => changeBreathMode('heartCoherence')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[0]}
        exitAnim={buttonsExitAnims[0]}
      />
    </ButtonContainer>
  );
};

const ButtonContainer = styled(motion.div)({
  display: 'flex',
  flexDirection: 'column-reverse',
});
