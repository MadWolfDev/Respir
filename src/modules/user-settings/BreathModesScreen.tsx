import { styled } from '@mui/material/styles';
import { useBreathModeStore } from '../../store/breathModeStore';
import { motion } from 'framer-motion';
import { Dispatch, SetStateAction } from 'react';
import { Display } from './WelcomeScreen';
import { BreathModes } from '../../store/defaultBreathConfigs';
import { BreathModesDisplay } from '../../store/BreathModesDisplay.type';
import { AnimVariant } from '../breath-activity/createBreathAnimationVariants';
import { CustomButton } from './CustomButton';
import {
  createButtonsExitAnim,
  createButtonsVariantAnim,
} from './createUIAnimation';

export const BreathModesScreen = ({
  setCurrentDisplay,
}: {
  setCurrentDisplay: Dispatch<SetStateAction<Display>>;
}) => {
  const updateBreathModeSelection = useBreathModeStore(
    (state) => state.updateBreathMode
  );
  const changeBreathMode = (newBreathMode: BreathModes) => {
    updateBreathModeSelection(newBreathMode);
    setCurrentDisplay(Display.slidersDisplay);
  };

  const buttonsExitAnims = createButtonsExitAnim(5);
  const buttonsAnimVariants = createButtonsVariantAnim(5);

  return (
    <ButtonContainer data-testid="mode-buttons">
      <CustomButton
        buttonText={BreathModesDisplay.custom}
        handleClick={() => changeBreathMode('custom')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[4]}
        exitAnim={buttonsExitAnims[4]}
      />

      <CustomButton
        buttonText={BreathModesDisplay.square}
        handleClick={() => changeBreathMode('square')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[3]}
        exitAnim={buttonsExitAnims[3]}
      />

      <CustomButton
        buttonText={BreathModesDisplay.relaxation}
        handleClick={() => changeBreathMode('relaxation')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[2]}
        exitAnim={buttonsExitAnims[2]}
      />

      <CustomButton
        buttonText={BreathModesDisplay.vitality}
        handleClick={() => changeBreathMode('vitality')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[1]}
        exitAnim={buttonsExitAnims[1]}
      />

      <CustomButton
        buttonText={BreathModesDisplay.heartCoherence}
        handleClick={() => changeBreathMode('heartCoherence')}
        buttonFixed={true}
        animVariant={buttonsAnimVariants[0]}
        exitAnim={buttonsExitAnims[0]}
      />
    </ButtonContainer>
  );
};

const ButtonContainer = styled(motion.div)({
  display: 'flex',
  flexDirection: 'column-reverse',
});
