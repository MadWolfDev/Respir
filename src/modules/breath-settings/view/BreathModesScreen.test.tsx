import { BreathModesScreen } from './BreathModesScreen';
import { renderJest } from '../../../libs/renderJest';

describe('BreathModesScreen', () => {
  it('should display all the buttons', () => {
    const setStateMock = jest.fn();

    const component = renderJest(
      <BreathModesScreen setCurrentDisplay={setStateMock} />
    );
    expect(component.baseElement).toMatchSnapshot();
  });
});
