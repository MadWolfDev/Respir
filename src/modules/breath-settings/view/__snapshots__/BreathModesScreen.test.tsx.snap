// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreathModesScreen should display all the buttons 1`] = `
<body>
  <div>
    <div
      class="css-1v3870x"
    >
      <div
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-pbsuc8-MuiButtonBase-root-MuiButton-root"
        role="button"
        style="transform: translateY(36.5px) translateZ(0);"
        tabindex="0"
      >
        Personnalisé
      </div>
      <div
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-pbsuc8-MuiButtonBase-root-MuiButton-root"
        role="button"
        style="transform: translateY(36.5px) translateZ(0);"
        tabindex="0"
      >
        Respiration carrée
      </div>
      <div
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-pbsuc8-MuiButtonBase-root-MuiButton-root"
        role="button"
        style="transform: translateY(36.5px) translateZ(0);"
        tabindex="0"
      >
        Respiration détente
      </div>
      <div
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-pbsuc8-MuiButtonBase-root-MuiButton-root"
        role="button"
        style="transform: translateY(36.5px) translateZ(0);"
        tabindex="0"
      >
        Respiration vitalité
      </div>
      <div
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedEnable MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-pbsuc8-MuiButtonBase-root-MuiButton-root"
        role="button"
        style="transform: translateY(36.5px) translateZ(0);"
        tabindex="0"
      >
        Cohérence cardiaque
      </div>
    </div>
  </div>
</body>
`;
